crypto/libsodium-fork/src/libsodium/crypto_sign/ed25519/ref10/batch.c

```diff
@@ -0,0 +1,318 @@
+#include "crypto_hash_sha512.h"
+#include "sign_ed25519_ref10.h"
+#include "private/ed25519_ref10.h"
+#include "private/ed25519_ref10_sc.h"
+#include "crypto_sign.h"
+
+
+#include <limits.h>
+#include <stdint.h>
+#include <string.h>
+
+
+void ed25519_randombytes_unsafe(void *p, size_t len);
```

Comment:
Where is this implemented? I only see test implementations — does there need to be one in non-test code that calls randombytes_buf?

Reply:
this function is implemented in the go code.
so we can use the same random source

Reply:
oh, OK. I didn't see anything from grepping for `ed25519_randombytes_unsafe` in the repo, but I guess `crypto_sign_ed25519_open_batch` is not yet wired up to Go yet